optuna/storages/_rdb/storage.py

```diff
@@ -14,34 +14,49 @@
 from typing import Optional
 from typing import Sequence
 from typing import Set
+from typing import TYPE_CHECKING
 import uuid
 
-import alembic.command
-import alembic.config
-import alembic.migration
-import alembic.script
 import numpy as np
-from sqlalchemy import func
-from sqlalchemy import orm
-from sqlalchemy.engine import create_engine
-from sqlalchemy.engine import Engine  # NOQA
-from sqlalchemy.exc import IntegrityError
-from sqlalchemy.exc import OperationalError
-from sqlalchemy.exc import SQLAlchemyError
-from sqlalchemy.sql import functions
 
 import optuna
 from optuna import distributions
 from optuna import version
+from optuna._imports import _LazyImport
 from optuna.storages._base import BaseStorage
 from optuna.storages._base import DEFAULT_STUDY_NAME_PREFIX
-from optuna.storages._rdb import models
 from optuna.study._study_direction import StudyDirection
 from optuna.study._study_summary import StudySummary
 from optuna.trial import FrozenTrial
 from optuna.trial import TrialState
 
 
+if TYPE_CHECKING:
+    import alembic.command as alembic_command
+    import alembic.config as alembic_config
+    import alembic.migration as alembic_migration
+    import alembic.script as alembic_script
+    import sqlalchemy
+    import sqlalchemy.engine as sqlalchemy_engine
+    import sqlalchemy.exc as sqlalchemy_exc
+    import sqlalchemy.orm as sqlalchemy_orm
+    import sqlalchemy.sql.functions as sqlalchemy_sql_functions
+
+    from optuna.storages._rdb import models as optuna_rdb_models
+else:
+    alembic_command = _LazyImport("alembic.command")
+    alembic_config = _LazyImport("alembic.config")
+    alembic_migration = _LazyImport("alembic.migration")
+    alembic_script = _LazyImport("alembic.script")
+
+    sqlalchemy = _LazyImport("sqlalchemy")
+    sqlalchemy_engine = _LazyImport("sqlalchemy.engine")
+    sqlalchemy_exc = _LazyImport("sqlalchemy.exc")
+    sqlalchemy_orm = _LazyImport("sqlalchemy.orm")
+    sqlalchemy_sql_functions = _LazyImport("sqlalchemy.sql.functions")
+
+    optuna_rdb_models = _LazyImport("optuna.storages._rdb.models")
```

Comment:
Not a strong opinion but how about both keeping the diff smaller and the code less verbose? I believe there are dittos elsewhere.
```suggestion
    models = _LazyImport("optuna.storages._rdb.models")
```